frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -8,16 +8,21 @@
  */
 
 import { OpPatch } from 'json-patch';
+import { decodeToken } from 'react-jwt';
 import helpers from '../../helpers';
-import { getToken, logout } from '../auth';
+import { getToken, logout, setToken } from '../auth';
 import { getCluster } from '../util';
 import { KubeMetrics, KubeObjectInterface } from './cluster';
+import { KubeToken } from './token';
 
 const BASE_HTTP_URL = helpers.getAppUrl();
 const BASE_WS_URL = BASE_HTTP_URL.replace('http', 'ws');
 const CLUSTERS_PREFIX = 'clusters';
 const JSON_HEADERS = { Accept: 'application/json', 'Content-Type': 'application/json' };
 const DEFAULT_TIMEOUT = 2 * 60 * 1000; // ms
+const MIN_LIFESPAN_FOR_TOKEN_REFRESH = 10; // sec
+
+var TOKEN_REFRESH_INPROGRESS = false;
```

Comment:
This shouldn't be all caps. All caps are used here just for constants.

Reply:
Also, maybe separate `INPROGRESS` -> `IN_PROGRESS`.

Reply:
let instead of var